server/portal/apps/workspace/api/views.py

```diff
@@ -218,47 +228,63 @@ def post(self, request, *args, **kwargs):
 
             # url encode inputs
-            if job_post['inputs']:
-                job_post = url_parse_inputs(job_post)
+            # TODOv3 update for v3
+            # if job_post['inputs']:
+            #     job_post = url_parse_inputs(job_post)
 
+            # TODOv3 potentially remove UserApplicationsManager but need to check if execution system needs keys
             # Get or create application based on allocation and execution system
             apps_mgr = UserApplicationsManager(request.user)
-            app = apps_mgr.get_or_create_app(job_post['appId'], job_post['allocation'])
-
-            if app.exec_sys:
-                return JsonResponse({"response": {"execSys": app.exec_sys.to_dict()}})
-
-            job_post['appId'] = app.id
+            print(apps_mgr)  # TODOv3 testing workaround
+            # app = apps_mgr.get_or_create_app(job_post['appId'], job_post['allocation'])
+
+            # TODOv3 update to determine if keys need to be pushed (as not using above code or replacement for abouve
+            # code: UserApplicationsManager get_or_create_app)
+            # if app.exec_sys:
+            #     return JsonResponse({"response": {"execSys": app.exec_sys.to_dict()}})
+
+            # TODOv3
+            if 'schedulerOptions' not in job_post['parameterSet']:
+                job_post['parameterSet']['schedulerOptions'] = []
+            job_post['parameterSet']['schedulerOptions'].append({"name": "Allocation",
+                                                                 "description": "The allocation associated with this job execution",
+                                                                 "include": True,
+                                                                 "arg": f"-A {job_post['allocation']}"})
             del job_post['allocation']
 
-            if settings.DEBUG:
-                wh_base_url = settings.WH_BASE_URL + '/webhooks/'
-                jobs_wh_url = settings.WH_BASE_URL + reverse('webhooks:jobs_wh_handler')
-            else:
-                wh_base_url = request.build_absolute_uri('/webhooks/')
-                jobs_wh_url = request.build_absolute_uri(reverse('webhooks:jobs_wh_handler'))
+            # TODOv3 Webhooks/notifications
+            #  if settings.DEBUG:
+            #     wh_base_url = settings.WH_BASE_URL + '/webhooks/'
+            #     jobs_wh_url = settings.WH_BASE_URL + reverse('webhooks:jobs_wh_handler')
+            # else:
+            #     wh_base_url = request.build_absolute_uri('/webhooks/')
+            #     jobs_wh_url = request.build_absolute_uri(reverse('webhooks:jobs_wh_handler'))
 
-            job_post['parameters']['_webhook_base_url'] = wh_base_url
-            job_post['notifications'] = [
-                {'url': jobs_wh_url,
-                 'event': e}
-                for e in settings.PORTAL_JOB_NOTIFICATION_STATES]
+            # TODOv3 webhook for wrapper (determine where this will go in the job (envVariables, right?)
```

Comment:
Yeah, the wh url will go in the job subscriptions object and as an `envVariable` like:
```
{
  "key": "_webhook_base_url",
  "value": jobs_wh_url
}
```

Reply:
👍